### PR TITLE
Add initial CODEOWNERS file

### PR DESCRIPTION
**Pull request by michaelpj**
Context: https://github.com/haskell/haskell-language-server/discussions/2588

The goal of this is to ensure that every plugin has at least one person associated with it who takes some responsibility for maintaining it over time.

The meaning of the CODEOWNERS file is: if you are listed on the RHS, you will be tagged for review on any PR touching the files on the LHS. Your review will not be required, so you won't have to look at every trivial change, but ideally you would be able to find time to look at any substantial changes.

**The users listed here are entirely my guesses, please do not be offended if you're not in there and you think you should be, or vice versa.** I've mostly just picked the original creator of the plugin, which isn't necessarily the right person. Please say if you do not want to be on the list, or if you think there is someone else who would want to be on the list. It's okay if we end up with some plugins that nobody claims responsibility for, at least then we know we have a problem :)

Additionally, my intention is to create a `hls-plugin-maintainers` Github team whose membership is the union of all the people listed here by plugins, so their attention can be drawn to e.g. GHC version compatibility work.

Mentioning all the people in this file for comment: @pepeiborra @drsooch @fendor @July541 @Ailrun @alanz @georgefst @jneira @tittoassini @berberman @eddiemundo @rayshih @OliverMadine @konn @isovector @wz1000 @Anton-Latukha @hasufell 

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2619"></a>

### CODE REVIEW

**Comment by michaelpj** [comment]
Perhaps I should be explicit: please approve this PR or comment approvingly if you're happy with how you appear in it (reactions are a bit ambiguous :) )

**Comment by konn** [comment]
Sorry for my being absent from HLS these days. I've been rather off-hand from HLS, but I'm still following (hopefully) most of GitHub issues / PRs mostly through e-mail. I will try at least to review new changes to Splice Plugin. Sorry again, and I would like to thank all the contributors who keep HLS alive!

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Core
+/ghcide @pepeiborra
+/hls-graph @pepeiborra
```
Comment (jneira) [inline CODEOWNERS:3]: miss `hie-compat` , `hls-plugin-api` and `hls-test-utils`

**Review by jneira — APPROVED** [review]
:100: this will make clear something you have to remember or getting via git blame. Thanks!

File: CODEOWNERS (hunk 1 above)
Comment (michaelpj) [reply CODEOWNERS:3]: `hie-compat` is further down. But yeah, I left off things I wasn't sure about. We don't *have* to have this for everything, I guess.

**Comment by georgefst** [comment]
This seems like a good idea. There have, in the past, been issues/PRs about `hls-fourmolu-plugin` where it would have been useful for me to be pinged sooner.

I'd also be happy to move it to the Fourmolu GitHub org if we decide to move away from the monorepo.

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/plugins/hls-stylish-haskell-plugin @Ailrun
+/plugins/hls-tactics-plugin @isovector
+
+# Benchmarking
+/shake-bench @pepeiborra
+
+# CI
+/.circleci @jneira @Anton-Latukha
+/.githhub @jneira @Anton-Latukha
+/.gitlab @jneira @hasufell
```
Comment (jneira) [inline CODEOWNERS:33]: miss the nix configuration

**Review by Anton-Latukha — APPROVED** [review]

**Review by konn — APPROVED** [review]

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Core
+/ghcide @pepeiborra
+/hls-graph @pepeiborra
+
+# Plugins
+/plugins/hls-alternate-number-format-plugin @drsooch
+/plugins/hls-brittany-plugin @fendor
```
Comment (fendor) [inline CODEOWNERS:7]: Is that an accident? I can't imagine I am the code-owner of hls-brittany-plugin? :O

**Review by fendor — CHANGES_REQUESTED** [review]
I would request a special case for `ghcide/session-loader` as I am very familiar with that code and I think my knowledge, prior work and tight integration with hie-bios is very valuable for that very complicated piece of code.

File: CODEOWNERS (hunk 3 above)
Comment (fendor) [reply CODEOWNERS:7]: Oh wow: https://github.com/haskell/haskell-language-server/pull/66

I did not remember that

**Review by drsooch — APPROVED** [review]

File: CODEOWNERS (hunk 3 above)
Comment (michaelpj) [reply CODEOWNERS:7]: It's fine to say "I don't want to be on the hook for it". If we end up with nobody there then we know we need to find someone!

File: CODEOWNERS (hunk 1 above)
Comment (jneira) [reply CODEOWNERS:3]: well i would add at least hls-plugin-api and hls-test-utils, feel free to add me to both (i think @pepeiborra is a good candidate :-P)
and well given hls-plugin-api is the interface with lsp, maybe you could be a really goo candidate too :-P
Same for hls-test-utils (interface with lsp-test)

And what about docs?, maybe you and me could be in charge

File: CODEOWNERS (hunk 3 above)
Comment (fendor) [reply CODEOWNERS:7]: I don't mind it, the code is trivial iirc and I did spend some work on it back HIE time, just don't want to take someone's work as my own. The very-original author seems to be @wz1000 according to https://github.com/haskell/haskell-ide-engine/pull/247/files

File: CODEOWNERS (hunk 1 above)
Comment (michaelpj) [reply CODEOWNERS:3]: I don't actually know anything about `hls-plugin-api` or `hls-test-utils`... yet. So I'd be happy to put myself in maybe when I know them better. But docs, sure.

Comment (jneira) [reply CODEOWNERS:3]: iirc @berberman did quite nice improvements around config in hls-plugin-api

File: CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+/plugins/hls-call-hierarchy-plugin @July541
+/plugins/hls-class-plugin @Ailrun
+/plugins/hls-eval-plugin @pepeiborra
+/plugins/hls-explicit-imports-plugin @pepeiborra
+/plugins/hls-floskell-plugin @alanz @Ailrun
+/plugins/hls-fourmolu-plugin @georgefst
+/plugins/hls-haddock-comments-plugin @berberman
+/plugins/hls-hlint-plugin @jneira
```
Comment (jneira) [inline CODEOWNERS:15]: i would add @eddiemundo if they agree, did a great work around recently

File: CODEOWNERS (hunk 1 above)
Comment (berberman) [reply CODEOWNERS:3]: Feel free to add me, though I was not being active as before, I still can take a look at these code. The initial code and idea are still in my mind :)

Comment (michaelpj) [reply CODEOWNERS:4]: Done.

File: CODEOWNERS (hunk 2 above)
Comment (michaelpj) [reply CODEOWNERS:40]: Added, with me and @berberman.

**Comment by michaelpj** [comment]
Added a few more:
- @fendor on `ghcide/session-loader`
- @michaelpj and @jneira on `docs`
- @michaelpj and @berberman on `*.nix`
- @jneira on `*.project`
- @jneira and @berberman on `hls-plugin-api`
- @jneira on `hls-test-utils`

**Review by fendor — APPROVED** [review]
Thanks!

File: CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+/hls-plugin-api @jneira @berberman
+/hls-test-utils @jneira
+/hie-compat @wz1000
+
+# Plugins
+/plugins/hls-alternate-number-format-plugin @drsooch
+/plugins/hls-brittany-plugin @fendor
+/plugins/hls-call-hierarchy-plugin @July541
+/plugins/hls-class-plugin @Ailrun
+/plugins/hls-eval-plugin @pepeiborra
```
Comment (pepeiborra) [inline CODEOWNERS:14]: I handed over the eval plugin to @tittoassini (https://github.com/haskell/haskell-language-server/pull/438#pullrequestreview-543322357)

```suggestion
/plugins/hls-eval-plugin @tittoassini
```

**Review by isovector — APPROVED** [review]

**Review by pepeiborra — APPROVED** [review]

File: CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+/plugins/hls-alternate-number-format-plugin @drsooch
+/plugins/hls-brittany-plugin @fendor
+/plugins/hls-call-hierarchy-plugin @July541
+/plugins/hls-class-plugin @Ailrun
+/plugins/hls-eval-plugin @tittoassini
+/plugins/hls-explicit-imports-plugin @pepeiborra
+/plugins/hls-floskell-plugin @alanz @Ailrun
+/plugins/hls-fourmolu-plugin @georgefst
+/plugins/hls-haddock-comments-plugin @berberman
+/plugins/hls-hlint-plugin @jneira
```
Comment (eddiemundo) [inline CODEOWNERS:19]: Yeah I agree.
```suggestion
/plugins/hls-hlint-plugin @jneira @eddiemundo
```

File: CODEOWNERS [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+/plugins/hls-module-name-plugin @tittoassini
+/plugins/hls-ormolu-plugin @alanz
+/plugins/hls-pragmas-plugin @berberman @Ailrun
```
Comment (eddiemundo) [inline CODEOWNERS:22]:
```suggestion
/plugins/hls-pragmas-plugin @berberman @Ailrun @eddiemundo
```

File: CODEOWNERS [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+# Benchmarking
+/shake-bench @pepeiborra
+
+# Docs
+/docs @michaelpj @jneira
+
+# CI
+/.circleci @jneira @Anton-Latukha
+/.githhub @jneira @Anton-Latukha
```
Comment (Ailrun) [inline CODEOWNERS:39]: There is a typo. BTW, I'm also willing to manage this, so:
```suggestion
/.github @jneira @Anton-Latukha @Ailrun
```

File: CODEOWNERS [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+/plugins/hls-fourmolu-plugin @georgefst
+/plugins/hls-haddock-comments-plugin @berberman
+/plugins/hls-hlint-plugin @jneira @eddiemundo
+/plugins/hls-module-name-plugin @tittoassini
+/plugins/hls-ormolu-plugin @alanz
+/plugins/hls-pragmas-plugin @berberman @Ailrun @eddiemundo
+/plugins/hls-qualify-imported-names-plugin @eddiemundo 
+/plugins/hls-refine-imports-plugin @rayshih 
+/plugins/hls-rename-plugin @OliverMadine
+/plugins/hls-retrie-plugin @pepeiborra
```
Comment (jneira) [inline CODEOWNERS:26]:
```suggestion
/plugins/hls-retrie-plugin @pepeiborra
/plugins/hls-selection-range-plugin @kokobd
```

**Review by jneira — CHANGES_REQUESTED** [review]
added new brand plugin by @kokobd, :1st_place_medal:

**Comment by michaelpj** [comment]
Nag: @tittoassini @alanz @July541 @rayshih @OliverMadine @hasufell

**Review by hasufell — APPROVED** [review]

**Review by kokobd — APPROVED** [review]

**Review by OliverMadine — APPROVED** [review]

**Comment by michaelpj** [comment]
To keep this moving, I'm going to remove the people who haven't assented and merge the rest. If they get back to this later then they can just add themselves with a PR! 

In a while, if we don't have anyone stepping up for the components that don't have anyone listed (`hie-compat`, `hls-eval-plugin`, `hls-module-name-plugin`, `hls-ormolu-plugin`, `hls-call-hierarchy-plugin`, `hls-refine-imports-plugin`), then we might want to put out a call for maintainers.

**Comment by georgefst** [comment]
> In a while, if we don't have anyone stepping up for the components that don't have anyone listed (`hie-compat`, `hls-eval-plugin`, `hls-module-name-plugin`, `hls-ormolu-plugin`, `hls-call-hierarchy-plugin`, `hls-refine-imports-plugin`), then we might want to put out a call for maintainers.

I'll happily take on Ormolu if no one else will.

**Pull request merged**